src/components/navbar/navbar.component.ts

```diff
@@ -1,24 +1,23 @@
 import {Component, OnInit} from '@angular/core';
-import {NavigationEnd, ROUTER_DIRECTIVES, Router} from '@angular/router';
+import {ROUTER_DIRECTIVES} from '@angular/router';
 import {Subscription} from 'rxjs/Rx';
 
 import {AuthenticationService, AuthenticationState} from '../../services/authentication.service';
+import {RoutingService} from '../../services/routing.service';
 import {UrlModalComponent} from '../urlModal/urlModal.component';
 
-const URL_BASE_PATH: string = 'choosetogo.hanaum.com/#';
-
 @Component({
   selector: 'nav-bar',
   template: require('./navbar.component.html'),
   styles: [require('./navbar.component.css')],
   directives: [ROUTER_DIRECTIVES, UrlModalComponent as any],
+  providers: [RoutingService]
```

Comment:
RoutingService should be provided in main.ts to make it a singleton. Remove all explicit providers.